src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,28 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
+    private final UserService userService;
+    private final DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper;
+
+    public UserController(UserService userService,
+                          DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper) {
+        this.userService = userService;
+        this.userDtoResponseMapper = userDtoResponseMapper;
+    }
+
+    @GetMapping("/users/by-email")
```

Comment:
I guess better use RequestMapping("/users") for class and 
method mapping ("/by-email")
But it is up to you